src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -73,6 +77,7 @@ public void ingest(
 
     fileStorageAdaptor.store(
         fileSize, mimeType, inputStream, datasetId, Map.of(FILE_NAME_METADATA_KEY, fileName));
+    fileStorageAdaptor.updateStatus(datasetId, STAGED);
```

Comment:
❓ Do storing and updating status need to happen as one unit of work?  In other words, would things break of the store happened but the status was never updated to staged?

Reply:
If that happened, you still wouldn't be able to retrieve the file or metacard. We only retrieve if the status is "stored". I do think your comment applies to the whole ingest flow. Ie it breaks after storing the file, but before storing the metacard. I'm not sure how we account for that.

Reply:
Typically we have been taking the approach of retrying. We shouldn't break, we would just be in a state where the dataset is not retrievable.

Reply:
Typically we have been relying on retries. Unfortunately yes, with these methods we could end up in a state where the file is retrievable, but the metacard is not, which is not desirable. I think the goal is that either all data in a dataset is retrievable, or none of it is. Unfortunately right now we have multiple Data sharing the same ID (the datasetId), which is abusing the DatasetId. Each data should have its own ID and a reference to the datasetId which tells it how it is related to other Data.